### ISSUE TITLE
Return rules field in SCA checks API endpoints

### ISSUE THREAD

**Issue by mgmacias95** [wazuh/wazuh#3055]
Hello team,

The @wazuh/core is adding a new field for SCA checks: _rules_. They're working on that in this issue: https://github.com/wazuh/wazuh/issues/2960.

The `GET/sca/:agent_id/checks/:check_id` endpoint must be updated to return the new field _rules_ too. Our unit tests must be updated too.

Best regards,
Marta

**Comment by mgmacias95** [comment]
I have already implemented this feature in branch [`dev-3.9-sca-checks-rules-api`](https://github.com/wazuh/wazuh/tree/dev-3.9-sca-checks-rules-api). I have created a fake DB to do some testing, these are some example outputs:
```javascript
# curl -u foo:bar "localhost:55000/sca/000/checks/cis_debian?pretty&q=id=5031"
{
   "error": 0,
   "data": {
      "totalItems": 1,
      "items": [
         {
            "title": "Ensure IP forwarding is disabled",
            "file": "/proc/sys/net/ipv4/ip_forward,/proc/sys/net/ipv6/ip_forward",
            "policy_id": "cis_debian",
            "result": "passed",
            "description": "The net.ipv4.ip_forward and net.ipv6.conf.all.forwarding flags are used to tell the system whether it can forward packets or not.",
            "remediation": "Set the following parameter in /etc/sysctl.conf or a /etc/sysctl.d/* file: net.ipv4.ip_forward = 0, net.ipv6.conf.all.forwarding = 0",
            "id": 5031,
            "rationale": "Setting the flags to 0 ensures that a system with multiple interfaces (for example, a hard proxy), will never be able to forward packets, and therefore, never serve as a router.",
            "compliance": [
               {
                  "key": "cis",
                  "value": "3.1.1"
               },
               {
                  "key": "cis_csc",
                  "value": "5.1"
               }
            ],
            "rules": [
               {
                  "type": "file",
                  "rule": "f:/proc/sys/net/ipv4/ip_forward -> 1;"
               },
               {
                  "type": "file",
                  "rule": "f:/proc/sys/net/ipv6/ip_forward -> 1;"
               }
            ]
         }
      ]
   }
}
# curl -u foo:bar "localhost:55000/sca/000/checks/cis_debian?pretty&q=id=5030"
{
   "error": 0,
   "data": {
      "totalItems": 1,
      "items": [
         {
            "policy_id": "cis_debian",
            "rationale": "The telnet protocol is insecure and unencrypted. The use of an unencrypted transmission medium could allow an unauthorized user to steal credentials. The ssh package provides an encrypted session and stronger security and is included in most Linux distributions.",
            "description": "The telnet package contains the telnet client, which allows users to start connections to other systems via the telnet protocol.",
            "file": "/etc/inetd.conf",
            "result": "passed",
            "id": 5030,
            "remediation": "Run the following command to uninstall telnet: # apt-get remove telnet",
            "title": "Ensure telnet client is not installed",
            "compliance": [
               {
                  "key": "cis_csc",
                  "value": "2.6, 4.5"
               },
               {
                  "key": "pci_dss",
                  "value": "2.2.3"
               },
               {
                  "key": "cis",
                  "value": "2.3.4"
               }
            ]
         }
      ]
   }
}
```

I'm waiting for @wazuh/core to finish their development to test this branch with real data.

**Comment by mgmacias95** [comment]
@Skeptor has shared the full database with all `sca_check_rules` rows populated. I have integrated it into our tests and I have tested it. It works correctly.

I have also fixed a bug in our implementation where filtering by any `compliance` or `rules` subfield the following way returned an error:
```javascript
# curl -u foo:bar "localhost:55000/sca/000/checks/cis_debian?pretty&q=rules.type!=file"
{
   "error": 1408,
   "message": "Field does not exist."
}
```
It was necessary to just use `type` (the name used in the DB underneath). This was quite confusing since the API returns the field nested:
```javascript
# curl -u foo:bar "localhost:55000/sca/000/checks/cis_debian?pretty&q=type!=file"
{
   "error": 0,
   "data": {
      "totalItems": 1,
      "items": [
         {
            "title": "Ensure inetd is not installed",
            "description": "The inetd daemon listens for well known services and dispatches the appropriate daemon to properly respond to service requests.",
            "rationale": "If there are no inetd services required, it is recommended that the daemon be removed.",
            "result": "failed",
            "policy_id": "cis_debian",
            "remediation": "Run the following commands to uninstall openbsd-inetd and inetutils-inetd: apt-get remove openbsd-inetd; apt-get remove inetutils-inetd",
            "id": 5018,
            "file": "/etc/inetd.conf",
            "compliance": [
               {
                  "key": "cis",
                  "value": "2.1.2"
               },
               {
                  "key": "pci_dss",
                  "value": "2.2.2"
               },
               {
                  "key": "cis_csc",
                  "value": "9.2"
               }
            ],
            "rules": [
               {
                  "type": "process",
                  "rule": "p:inetd;"
               }
            ]
         }
      ]
   }
}
```
